Assignments/SCF-Session-2/hexadecimalCalc/HexCalc.java

```diff
@@ -0,0 +1,277 @@
+	    	return hex;  
+	}
+	
+	public boolean greater(String hexNum1, String hexNum2)
+	{
+		hexNum1 = removeZeros(hexNum1);
+		hexNum2 = removeZeros(hexNum2);
+		boolean greaterNum  = false;
+		String digits = "0123456789ABCDEF";
+		hexNum1 = hexNum1.toUpperCase();
+		hexNum2 = hexNum2.toUpperCase();
+		
+		if(hexNum1.length() > hexNum2.length())
+			greaterNum = true;
+		else 
+			if(hexNum1.length() == hexNum2.length())
+		    	{
+				for(int i = 0; i < hexNum1.length(); i++)
+				{
+					if(digits.indexOf(hexNum1.charAt(i)) > digits.indexOf(hexNum2.charAt(i)))
+					{
+						greaterNum = true;
+						break;
+					}
+					else if(hexNum1.charAt(i) == hexNum2.charAt(i))
+						continue;
+					else
+						break;
+				}
+		    	}
+			return greaterNum;
+	}
+	
+	public boolean smaller(String hexNum1, String hexNum2)
+	{
+		hexNum1 = removeZeros(hexNum1);
+		hexNum2 = removeZeros(hexNum2);
+		boolean smallerNum  = false;
+		String digits = "0123456789ABCDEF";
+		hexNum1 = hexNum1.toUpperCase();
+		hexNum2 = hexNum2.toUpperCase();
+		
+		if(hexNum1.length() < hexNum2.length())
+			smallerNum = true;
+		else 
+			if(hexNum1.length() == hexNum2.length())
+		    	{
+				for(int i = 0; i < hexNum1.length(); i++)
+				{
+					if(digits.indexOf(hexNum1.charAt(i)) < digits.indexOf(hexNum2.charAt(i)))
+					{
+						smallerNum = true;
+						break;
+					}
+					else if(hexNum1.charAt(i) == hexNum2.charAt(i))
+						continue;
+					else
+						break;
+				}
+		    	}
+			return smallerNum;
+	}
+	
+	public boolean equality(String hexNum1, String hexNum2)
+	{
+		hexNum1 = removeZeros(hexNum1);
+		hexNum2 = removeZeros(hexNum2);
+		boolean equal = false;
+		
+		if(hexNum1.equals(hexNum2))
+			equal = true;
+		return equal;
+	}
+	
+	public String removeZeros(String hexNum)
+	{
+		int i = 0;
+		String newString = "";
+		while(hexNum.charAt(i) == '0')
+			i++;
+		while(i	< hexNum.length())
+		{
+			newString += hexNum.charAt(i);
+			i++;
+		}
+		
+		return newString;
+	}
+}
+
+
+public class HexCalc 
+{
+	public static void main(String[] args)
+	{
+		int decNum, choice;
+		String hexNum, hexNum1, hexNum2;
+		
+		Functions fun = new Functions();
+		
+		Scanner sc = new Scanner(System.in);
+		
+		do
+		{
+			System.out.println("MENU");
+			System.out.println("1. Add");
+			System.out.println("2. Substract");
+			System.out.println("3. Multiply");
+			System.out.println("4. Divide");
+			System.out.println("5. Hexadecimal to Decimal");
+			System.out.println("6. Decimal to Hexadecimal");
+			System.out.println("7. Greater Hexadecimal number");
+			System.out.println("8. Smaller Hexadecimal number");
+			System.out.println("9. Check for Equal Hexadecimal number");
+			System.out.println("10. Exit");
+			System.out.println("Enter your choice:");
+			choice = sc.nextInt();
+		
+			switch(choice)
+			{
+				case 1: System.out.println("Enter two hexadecimal numbers:");
+					hexNum1 = sc.next();
+					hexNum2 = sc.next();
+					String hexAdd = fun.add(hexNum1, hexNum2);
```

Comment:
No need to declare `hexAdd` if no further use. Statement can be written as `System.out.println("Addition:" +fun.add(hexNum1, hexNum2));`

Reply:
done